core/storage/user/gae_models.py

```diff
@@ -672,3 +672,119 @@ def construct_model_id(cls, user_id, skill_id):
+
+    @classmethod
+    def get_all_scores_of_user(cls, user_id):
+        """Gets all scores for a given user.
+
+        Args:
+            user_id: str. The id of the user.
+
+        Returns:
+            list(UserContributionsScoringModel). All instances for the given
+                user.
+        """
+        return cls.get_all().filter(cls.user_id == user_id).fetch()
+
+    @classmethod
+    def get_all_users_with_score_above_minimum_for_category(
+            cls, score_category):
+        """Gets all instances which have score above the
+        MINIMUM_SCORE_REQUIRED_TO_REVIEW threshold for the given category.
+
+        Args:
+            score_category: str. The category being queried.
+
+        Returns:
+            list(UserContributionsScoringModel). All instances for the given
+                category with scores above MINIMUM_SCORE_REQUIRED_TO_REVIEW.
+        """
+        return cls.get_all().filter(
+            cls.score_category == score_category).filter(
+                cls.score >= feconf.MINIMUM_SCORE_REQUIRED_TO_REVIEW).fetch()
+
+    @classmethod
+    def _get_instance_id(cls, user_id, score_category):
+        """Generates the instance id in the form {{score_category}}.{{user_id}}.
+
+        Args:
+            user_id: str. The ID of the user.
+            score_category: str. The category of suggestion to score the user
+                on.
+
+        Returns:
+            str. The instance ID for UserContributionScoringModel.
+        """
+        return '.'.join([score_category, user_id])
+
+    @classmethod
+    def get_score_of_user_for_category(cls, user_id, score_category):
+        """Gets the score of the user for the given score category.
+
+        Args:
+            user_id: str. The ID of the user.
+            score_category: str. The category of suggestion to score the user
+                on.
+
+        Returns:
+            float|None. The score of the user in the given category.
+        """
+        instance_id = cls._get_instance_id(user_id, score_category)
+        model = cls.get_by_id(instance_id)
+
+        return model.score if model else None
+
+    @classmethod
+    def create(cls, user_id, score_category, score):
+        """Creates a new UserContributionScoringModel entry.
+
+        Args:
+            user_id: str. The ID of the user.
+            score_category: str. The category of the suggestion.
+            score: float. The score of the user.
+
+        Raises:
+            Exception: There is already an entry with the given id.
+        """
+        instance_id = cls._get_instance_id(user_id, score_category)
+
+        if cls.get_by_id(instance_id):
+            raise Exception('There is already an entry with the given id: %s' %
+                            instance_id)
+
+        cls(id=instance_id, user_id=user_id, score_category=score_category,
+            score=score).put()
+
+    @classmethod
+    def update_score_for_user(cls, user_id, score_category, update_by):
```

Comment:
Nit on naming choice: this sounds like it's replacing the score, not incrementing it. Consider maybe:

``increment_score_for_user`` and s/``update_by``/``score_change``/ or something similar.

Reply:
Done